pkgs/flarum/composer-env.nix

```
@@ -0,0 +1,244 @@
+# This file originates from composer2nix
+
+{ stdenv, lib, writeTextFile, fetchurl, php, unzip, phpPackages }:
+
+let
+  inherit (phpPackages) composer;
+
+  filterSrc = src:
+    builtins.filterSource (path: type: type != "directory" || (baseNameOf path != ".git" && baseNameOf path != ".git" && baseNameOf path != ".svn")) src;
```

This code is generic, and while usually I'd say that stuff should be part of `composer2nix` and just be `import`ed, you can actually trick it into doing what you want with something along these lines:

```suggestion
    let
      flarum-src = builtins.fetchTarball ...;
      # hardcode flarum source here. not great
      filtered-src = builtins.filterSource (path: type: type != "directory" || (baseNameOf path != ".git" && baseNameOf path != ".git" && baseNameOf path != ".svn")) flarum-src;
    # combine what with what's actually in this source tree
    in pkgs.symlinkJoin [ src filtered-src ];
```

Obviously this is fugly, but it doesn't make any sense to touch the generated sources file. I'm sure you'll find a more reasonable workaround.

No idea if this will actually work with symlinks.